hooks/push

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+set -ex
+
+LASTEST_MAJOR="3"
+DRIVERS="mssql mysql psql"
+
+# inspired from https://github.com/fsaintjacques/semver-tool
+SEMVER_REGEX="^[vV]?(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)(\-(0|[1-9][0-9]*|[0-9]*[A-Za-z-][0-9A-Za-z-]*)(\.(0|[1-9][0-9]*|[0-9]*[A-Za-z-][0-9A-Za-z-]*))*)?(\+[0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*)?$"
+
+if [[ "${SOURCE_BRANCH}" =~ ${SEMVER_REGEX} ]]; then
+    major="${BASH_REMATCH[1]}"
+    minor="${BASH_REMATCH[2]}"
+    patch="${BASH_REMATCH[3]}"
+    prere="${BASH_REMATCH[4]}"
+else
+    echo "The tag is not semantic versioning 2.0.0 compatible, not pushing any build."
+    exit 1
+fi
+
+# tag and push a docker image
+function tagAndPush {
+    src_tag="$1"
+    dst_tag="$2"
+
+    docker tag "$src_tag" "$dst_tag"
+    docker push "$dst_tag"
+}
+
+# tag and push a version of each image (main image, and single driver images)
+function tagAndPushVersion {
+    version="$1"
+
+    # main image (all drivers)
+    tagAndPush sqlboiler "${DOCKER_REPO}:${version}"
+
+    # single driver images
+    for driver in $DRIVERS; do
+        tagAndPush "sqlboiler-${driver}" "${DOCKER_REPO}:${version}-${driver}"
+    done
+}
+
+# if not a prerelease
+if [[ "${prere}" == "" ]]; then
+    tagAndPushVersion "${major}"
```

Comment:
Just asking the question: Given that the entire idea is to lock users to a specific version of the tool else they will likely get different generated outputs and trash against each other in git, is this tagging scheme something we want to support?

It seems much like the Go modules decision of "always try to get the latest X" versus "get the latest required". If you go with the former you can pull changes you didn't expect/want because you didn't get a chance to look at them and make a decision to upgrade. If you do the latter you have to remember to decide to upgrade.

For sqlboiler, what makes the most sense?

Reply:
Personally, I would leave this decision to the user. It's fairly common practice in docker hub to tag each major version and it's up to the user to decide what's the best for their environment. Some user could for example want to have their CI always using the latest v3 to make sure there isn't any incompatibility, and I'm sure there is a lot of use-cases that I cannot think about.